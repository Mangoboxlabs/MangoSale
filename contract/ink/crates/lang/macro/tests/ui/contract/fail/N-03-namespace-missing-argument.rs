use ink_lang as ink;

#[ink::contract]
mod invalid_namespace_identifier {
    #[ink(storage)]
    pub struct MyStorage {}

    #[ink(namespace)]
    impl MyStorage {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
