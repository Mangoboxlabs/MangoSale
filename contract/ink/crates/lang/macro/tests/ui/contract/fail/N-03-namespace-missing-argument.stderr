error: encountered #[ink(namespace)] that is missing its string parameter. Did you mean #[ink(namespace = name: str)] ?
 --> $DIR/N-03-namespace-missing-argument.rs:8:11
  |
8 |     #[ink(namespace)]
  |           ^^^^^^^^^
